draft-ietf-masque-h3-datagram.md

```diff
@@ -359,23 +359,35 @@ violation of these requirements MUST treat the HTTP message as malformed.
 
 ## Error Handling
 
-When an error occurs in processing the Capsule Protocol, the receiver MUST treat
-the message as malformed or incomplete, according to the underlying transport
-protocol. For HTTP/3, the handling of malformed messages is described in
-{{Section 4.1.3 of H3}}. For HTTP/2, the handling of malformed messages is
+Unless otherwise stated, Capsule Protocol endpoints are expected to apply the
+same message format rules as the underlying HTTP version. When a receiver
+encounters a processing error, it MUST be treated as if it were a malformed or
+incomplete message. For HTTP/3, the handling of malformed messages is described
+in {{Section 4.1.3 of H3}}. For HTTP/2, the handling of malformed messages is
 described in {{Section 8.1.1 of H2}}. For HTTP/1.1, the handling of incomplete
 messages is described in {{Section 8 of H1}}.
 
+For HTTP/2 and HTTP/3, once both endpoints agree to use the Capsule Protocol,
+the frame usage requirements of the stream change. In HTTP/2, only DATA or
+stream management frames (RST_STREAM, WINDOW_UPDATE, and PRIORITY) are
+permitted; other frame types MUST NOT be sent and MUST be treated as a stream
+error ({{Section 5.4.2 of H2}}) if received. In HTTP/3, only DATA frames are
+permitted; other frame types MUST NOT be sent and MUST be treated as connection
+error of type H3_FRAME_UNEXPECTED ({{Section 8 of H3}}) if received. In both
+cases, extension frame types MAY be used if specifically permitted by the
+definition of the extension.
```

Comment:
This text duplicates normative text from our references, how about the below? Also, I'd move it to the end of `{{capsule-protocol}}` since it isn't specific to error handling.

```suggestion
Since the Capsule Protocol only applies to definitions of new HTTP Upgrade
Tokens, in HTTP/2 and HTTP/3 it can only be used with the CONNECT method.
Therefore, once both endpoints agree to use the Capsule Protocol, the frame
usage requirements of the stream change as specified in {{Section 8.5 of H2}}
and {{Section 4.2 of H3}}.
```

Reply:
Oh yes, that is much better